content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+---
+layout: post
+title: "Refactoring Conditional BuildSteps into Jenkins Pipeline"
+tags:
+- pipeline
+- freestyle
+- plugins
+- conditional-build-step
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:http://cloudbees.com[CloudBees].
+
+== Introduction
+
+With all the new developments in Jenkins Pipeline (and
+link:https://jenkins.io/blog/2017/01/12/declarative-pipeline-beta-2/[Declarative Pipeline on the horizon]),
+it's easy to forget what we did to create "pipelines" before
+*link:https://jenkins.io/doc/book/pipeline/[Pipeline]*.
+There are number of plugins, some that have been around since before Jenkins was
+Jenkins, that enable users to create "pipelines" in Jenkins.
+For example, basic job chaining worked well in many cases, and the
+link:https://plugins.jenkins.io/parameterized-trigger[Parameterized Trigger plugin]
+made chaining more flexible.
+However, creating chained jobs with conditional behavior was
+still one of the harder things to do in Jenkins.
+
+The
+link:https://plugins.jenkins.io/conditional-buildstep[Conditional BuildStep plugin]
+is a powerful tool that has allowed Jenkins users to write Jenkins jobs with
+complex conditional logic.  The plugin also includes a set of
+token macros that expose a wide range of job state information on which to base
+those conditions.
+
+In this post, we'll take a look at how we might migrate Freestyle jobs with
+Conditional BuildSteps to Jenkins Pipeline.
+Unlike Freestyle jobs, implementing conditional behavior in Jenkins Pipeline is trivial,
+but migrating jobs using complex conditional build steps will require a bit more care.
+
+
+== Graphical Programming
+
+The Conditional BuildStep plugin lets users add conditional logic to Freestyle
+jobs from within the Jenkin web UI.  It does this by:
+
+* Adding two types of Conditional BuildStep ("Single" and "Multiple") -
+these build steps contain one or more other build steps to be run when the configured
+condition is met
+* Adding a set of Condition operations -
+these control whether the Conditional BuildStep execute the contained step(s)
+* Leveraging the Token Macro facility -
+these provide values to the Conditions for evaluation
+
+In the example below, this project will run the shell script step when the value of the
+`REQUESTED_ACTION` token equals "greeting".
+
+image::/images/post-images/2017-01-17/freestyle-conditional-param.png[Freestyle Job Parameters, role="center", width=800]
```

Comment:
Don't forget to update the image URLs to match the current filenames (2017-01-17 -> 2017-01-18).